### ISSUE TITLE
Can't link MathColorTest when building Magnum

### ISSUE THREAD

**Issue by LB--** [mosra/magnum#109]
In relation to https://github.com/mosra/magnum/issues/104#issuecomment-137994940

I'm using Windows 10 64bit and am compiling with [nuwen MinGW](http://nuwen.net/mingw.html) GCC 5.1.0, which is 64-bit. I built and installed Corrade without issue, but am having issues building Magnum.

`cmake -G "Unix Makefiles" ../ -DCMAKE_INSTALL_PREFIX=C:/Corrade -DBUILD_TESTS=ON -DBUILD_GL_TESTS=ON` after which I just run `make`. It builds Magnum and starts building the tests, then when it gets to MathColorTest, this happens:
```txt
Scanning dependencies of target MathColorTest
[ 20%] Building CXX object src/Magnum/Math/Test/CMakeFiles/MathColorTest.dir/ColorTest.cpp.obj
[ 20%] Linking CXX executable MathColorTest.exe
CMakeFiles/MathColorTest.dir/objects.a(ColorTest.cpp.obj):ColorTest.cpp:(.text$_ZN7Corrade7Utility18ConfigurationGroup8setValueIN6Magnum4Math6Color3IfEEEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKT_jNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE[_ZN7Corrade7Utility18ConfigurationGroup8setValueIN6Magnum4Math6Color3IfEEEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKT_jNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE]+0x31): undefined reference to `Corrade::Utility::ConfigurationValue<Magnum::Math::Vector<3ull, float> >::toString(Magnum::Math::Vector<3ull, float> const&, Corrade::Containers::EnumSet<Corrade::Utility::ConfigurationValueFlag, (unsigned char)255>)'
CMakeFiles/MathColorTest.dir/objects.a(ColorTest.cpp.obj):ColorTest.cpp:(.text$_ZNK7Corrade7Utility18ConfigurationGroup5valueIN6Magnum4Math6Color3IfEEEET_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEjNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE[_ZNK7Corrade7Utility18ConfigurationGroup5valueIN6Magnum4Math6Color3IfEEEET_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEjNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE]+0x89): undefined reference to `Corrade::Utility::ConfigurationValue<Magnum::Math::Vector<3ull, float> >::fromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Corrade::Containers::EnumSet<Corrade::Utility::ConfigurationValueFlag, (unsigned char)255>)'
CMakeFiles/MathColorTest.dir/objects.a(ColorTest.cpp.obj):ColorTest.cpp:(.text$_ZN7Corrade7Utility18ConfigurationGroup8setValueIN6Magnum4Math6Color4IfEEEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKT_jNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE[_ZN7Corrade7Utility18ConfigurationGroup8setValueIN6Magnum4Math6Color4IfEEEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKT_jNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE]+0x31): undefined reference to `Corrade::Utility::ConfigurationValue<Magnum::Math::Vector<4ull, float> >::toString(Magnum::Math::Vector<4ull, float> const&, Corrade::Containers::EnumSet<Corrade::Utility::ConfigurationValueFlag, (unsigned char)255>)'
CMakeFiles/MathColorTest.dir/objects.a(ColorTest.cpp.obj):ColorTest.cpp:(.text$_ZNK7Corrade7Utility18ConfigurationGroup5valueIN6Magnum4Math6Color4IfEEEET_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEjNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE[_ZNK7Corrade7Utility18ConfigurationGroup5valueIN6Magnum4Math6Color4IfEEEET_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEjNS_10Containers7EnumSetINS0_22ConfigurationValueFlagELh255EEE]+0x8d): undefined reference to `Corrade::Utility::ConfigurationValue<Magnum::Math::Vector<4ull, float> >::fromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Corrade::Containers::EnumSet<Corrade::Utility::ConfigurationValueFlag, (unsigned char)255>)'
collect2.exe: error: ld returned 1 exit status
src/Magnum/Math/Test/CMakeFiles/MathColorTest.dir/build.make:99: recipe for target 'src/Magnum/Math/Test/MathColorTest.exe' failed
make[2]: *** [src/Magnum/Math/Test/MathColorTest.exe] Error 1
CMakeFiles/Makefile2:910: recipe for target 'src/Magnum/Math/Test/CMakeFiles/MathColorTest.dir/all' failed
make[1]: *** [src/Magnum/Math/Test/CMakeFiles/MathColorTest.dir/all] Error 2
makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```
I'm terrible with liker errors, but it looks like something from Corrade is not being found. Any ideas?

**Comment by mosra** [comment]
Huh, just double checked on MinGW-w64 GCC 5.2 (though crosscompiling from Linux) and all works here. I recall that I stumbled upon some linker issues around 5.0, but everything was working again with 5.2 so I didn't bother (and then forgot).

Building 5.1 now so I can test on that as well.

**Comment by mosra** [comment]
Hmm.

So, I compiled MinGW-w64 GCC 5.1.0 on my Linux machine. Nope, couldn't reproduce. Then I downgraded MinGW-w64 binutils to 2.25 which was in the nuwen distro, nope, no change either. Then I tried to downgrade MinGW-w64 CRT to 4.0.2 to have *really everything* on the same version, nope, still got nothing. So I gave up and rebooted to Windows 8.0.

Installed nuwen distro, cloned the repos, created clean build dirs, compiled Magnum and *bam*, there it was. So I thought that yeah, that surely wasn't there with previous GCC version. So I downloaded previous version of the nuwen distro, compiled Corrade, but Magnum compilation of course failed because 64bit MinGW-w64 GCC 4.9.2 is producing binaries that are crashing on startup -- in C++11 mode there's broken `std::move(std::string)`, everyone knows that, right?

I immediately gave up trying to get nuwen distro compile in 32bit mode because it once proved to be impossible for my tiny brain and tried to install *official* i686 MinGW distro. Then I got lost on SourceForge trying to find something that is able to install *anything*. Fortunately there are those tiny graphs that show which files the people are interested in and which not, so I downloaded the file that had the bluest graph, waited until it downloaded one and half universe, shuffled some files around, renamed `mingw32-make.exe` to `make.exe` so CMake would stop crying and then it finally started building. *Bam*, there was that linker error again.

Then I got **mad** at this pathetic platform which doesn't have a *single* compiler that does what one would expect. I installed proper IDE, fixed the linker error simply by giving up on MinGW because I hate that mess (86c8fdef8d69830fa8b5333f5883209c2ea989b1) and then proceeded to finally *work around all [absurd and ridiculous bugs](https://github.com/mosra/magnum/compare/8c2eaea406b35dcf9caa41cddc6eb845ca052f82...69949b991ad308e374a1c395f9f957a4de65eac7) in MSVC 2015* and also *all* remaining issues and failing tests (except for the GL ones) on Windows so *all* the builds are finally [![green](https://ci.appveyor.com/api/projects/status/5b477m034cfaskse/branch/master?svg=true)](https://ci.appveyor.com/project/mosra/magnum/branch/master).

*Phew.*

Build on MinGW is now fixed and thanks in advance for the GL test output :)
